src/query/ts/m3db/encoded_step_iterator_test.go

```diff
@@ -509,15 +510,47 @@ func setupBlock(b *testing.B, iterations int, t iterType) (block.Block, reset) {
 	}
 
 	usePools := t == stepParallel
+
+	opts := newTestOptions()
+	if usePools {
+		poolOpts := xsync.NewPooledWorkerPoolOptions()
+		readWorkerPools, err := xsync.NewPooledWorkerPool(1024, poolOpts)
+		require.NoError(b, err)
+		readWorkerPools.Init()
+		opts = opts.SetReadWorkerPool(readWorkerPools)
+	}
+
+	for _, reset := range itersReset {
+		reset()
+	}
+
+	block, err := NewEncodedBlock(iters, models.Bounds{
+		Start:    start,
+		StepSize: stepSize,
+		Duration: window,
+	}, false, block.NewResultMetadata(), opts)
+
+	require.NoError(b, err)
+	return block, func() {
+			for _, reset := range itersReset {
+				reset()
+			}
+		},
+		setupProf(usePools, iterations)
+}
+
+func setupProf(usePools bool, iterations int) stop {
```

Comment:
The latest refactor to this file no longer supported profiling since the `defer p.Stop()` was being done in the `setupBlocks` subfunction, so this fixes that.